### PR TITLE
UI changes and optimizations

### PR DESCRIPTION
**Pull request by rsiebert**
This PR improves the UI and contains internal optimizations
- The UI has been improved and ported to the halo styles
- The main view contains tabs for fast access to channels and recordings
- The actionbar is used throughout all screens
- More programs will automatically be loaded when the user reaches the end of program list
- The recordings are grouped into tabs
- Orientation changes remember the selected tab and items
- and much more...

Note: 
- This PR contains the PR #19, #20, #21 and #22.
- The miminum android SDK version is now 11.

### CODE REVIEW

File: src/org/tvheadend/tvhguide/ChannelListTabsActivity.java [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+
+            @Override
+            public void onTabUnselected(Tab tab, FragmentTransaction ft) {
+                // Save the position of the tab that has been unselected.
+                prevTabPosition = tab.getPosition();
+                
+                // Detach the channel list fragment, because another will be attached
+                Fragment prevFragment = getFragmentManager().findFragmentByTag(tab.getText().toString());
+                if (prevFragment != null) {
+                    ft.detach(prevFragment);
```
Comment (kecksk0enig) [inline src/org/tvheadend/tvhguide/ChannelListTabsActivity.java:73]: detach ist still API-Level 13!
http://developer.android.com/reference/android/app/FragmentTransaction.html#detach%28android.app.Fragment%29

Comment (rsiebert) [inline src/org/tvheadend/tvhguide/ChannelListTabsActivity.java:73]: Thx for the information. I have changed the min SDK version to 13.

**Comment by stv0g** [comment]
Wow, that looks fantastic.

Do you want to provide us with compiled APK in the gh-pages branch for example?
I've no Android SDK setup on my systems..

Thank you :-)

**Comment by john-tornblom** [comment]
Yes, very nice indeed! Unfortunately, I'm extremely busy these days so it might take a while until I can have a look at the code. In addition, I'm still using Android 2.3 so I can't really test it thoroughly. Is it at all possible to keep the code backwards compatible? If not, I guess it gives me a reason to by some new hardware :)

**Comment by ghost** [comment]
Welcome
Can you add Polish language to the application and issue a version of the android market? The other issue I added Polish translation. Thank you.
Sorry for off topic :)

**Comment by kecksk0enig** [comment]
@john-tornblom
I think there is no way to keep it backwardcompatible without loosing lot of new gui features. Maybe there are some emulation layer one coulde use to get an almost equal looking and feeling feature but it would really bloat the code.
Are thhere any statistics how many useres of tvhguide are still on androd 2.x?
I currently do run a custom buildd of tvhguide iincluding those changes from rsiebert and my own channellogo update and it works really well.

@piterEL
If you are a nativespeaker of polish feel free to get a copy of values.xml and provide an polish translation by your self. You won´t even need a fully ide, just a texteditor would work.

**Comment by ghost** [comment]
I translated xml i this issue
https://github.com/john-tornblom/TVHGuide/pull/24

**Comment by rsiebert** [comment]
@PiterEL I have added the polish translation to the development and release branch. But there are many more strings that need translation.

@john-tornblom It could be possible to use the sherlock actionbar to support android 2.2 but I don't know how much effort this would be. It would be nice to see how many people still use this app with android 2.2.

@all I have added an unsigned apk (version 1.7.1) to the gh-pages branch.

**Comment by ghost** [comment]
Which strings ? Sorry i am beginner :) Tell me what isn't translated?

**Comment by ghost** [comment]
When can we download from google play version 1.7.1 ? Tomorrow i transleted to polish other string.xml

**Comment by john-tornblom** [comment]
@rsiebert  About 8% of all active installs are on Android 2.x. Google have [a tutorial on how to keep backwards comparability](http://developer.android.com/training/backward-compatible-ui). I'm not sure if it's feasible, what do you think?

**Comment by rsiebert** [comment]
@john-tornblom I think I'll give it a shot. I was busy implementing the multiple account functionality. Porting to android 2.2 is now on the list.

**Comment by rsiebert** [comment]
Maybe I should start squashing these commits. :-)

**Comment by john-tornblom** [comment]
@rsiebert great! I got a few minutes over last weekend and had a quick look. Apparently, it is possible to use external libs (provided by google), see http://developer.android.com/tools/support-library/setup.html for the details.

**Comment by rsiebert** [comment]
@john-tornblom New TVHGuide version 1.8 is out. I have ported the application to be compatible with android 2.2. It uses the android support library and the preference fragment from https://github.com/kolavar/android-support-v4-preferencefragment. Works great on the HTC mini with Android 2.2 and on mine with 4.3.

**Comment by ghost** [comment]
I installed your version(today) and i have a problem. I can't watch TV. When i click on play i have a ERROR OPENING FILE 66and other numer for example 5 , 77. On oficial app from GOOGLE PLAY i can watch TV with no problem... Please HELP

**Comment by rsiebert** [comment]
I'll check this. You can change in the settings that an external media player shall be used.

**Comment by ghost** [comment]
I use an external player name BSplayer and it works great on oficial TVHguide from google play.

**Pull request closed without merging**